airflow/ti_deps/dep_context.py

```diff
@@ -88,43 +88,85 @@ def __init__(
 
 
 # In order to be able to get queued a task must have one of these states
-QUEUEABLE_STATES = {
-    State.FAILED,
+SCHEDULEABLE_STATES = {
     State.NONE,
+    State.UP_FOR_RETRY,
+    State.UP_FOR_RESCHEDULE,
+}
+
+RUNNABLE_STATES = {
+    # For cases like unit tests and run manually
+    State.NONE,
+    State.UP_FOR_RETRY,
+    State.UP_FOR_RESCHEDULE,
+    # For normal scheduler/backfill cases
     State.QUEUED,
+}
+
+QUEUEABLE_STATES = {
     State.SCHEDULED,
-    State.SKIPPED,
-    State.UPSTREAM_FAILED,
-    State.UP_FOR_RETRY,
+}
+
+BACKFILL_QUEUEABLE_STATES = {
+    # For cases like unit tests and run manually
+    State.NONE,
     State.UP_FOR_RESCHEDULE,
+    State.UP_FOR_RETRY,
+    # For normal backfill cases
+    State.SCHEDULED,
 }
 
-# Context to get the dependencies that need to be met in order for a task instance to
-# be backfilled.
-QUEUE_DEPS = {
-    NotRunningDep(),
```

Comment:
Could/should we delete NotRunningDep and NotSkippedDep deps? They don't appear to be used anymore?  (Or should they still be used somewhere?)

Reply:
Good point, I don't see a use case for them.